app/src/main/java/mozilla/lockbox/store/FingerprintStore.kt

```diff
@@ -0,0 +1,206 @@
+package mozilla.lockbox.store
+
+import android.hardware.fingerprint.FingerprintManager
+import android.os.CancellationSignal
+import android.security.keystore.KeyGenParameterSpec
+import android.security.keystore.KeyPermanentlyInvalidatedException
+import android.security.keystore.KeyProperties
+import io.reactivex.Observable
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.rxkotlin.addTo
+import io.reactivex.subjects.PublishSubject
+import mozilla.lockbox.action.FingerprintSensorAction
+import mozilla.lockbox.extensions.filterByType
+import mozilla.lockbox.flux.Dispatcher
+import java.io.IOException
+import java.security.InvalidAlgorithmParameterException
+import java.security.InvalidKeyException
+import java.security.KeyStore
+import java.security.KeyStoreException
+import java.security.NoSuchAlgorithmException
+import java.security.NoSuchProviderException
+import java.security.UnrecoverableKeyException
+import java.security.cert.CertificateException
+import javax.crypto.Cipher
+import javax.crypto.KeyGenerator
+import javax.crypto.NoSuchPaddingException
+import javax.crypto.SecretKey
+
+open class FingerprintStore(
+    val dispatcher: Dispatcher = Dispatcher.shared
+) {
+    internal val compositeDisposable = CompositeDisposable()
+    private lateinit var fingerprintManager: FingerprintManager
+
+    private lateinit var keyStore: KeyStore
+    private lateinit var keyGenerator: KeyGenerator
+    private var cancellationSignal: CancellationSignal? = null
+    private var selfCancelled = false
+
+    private val _state: PublishSubject<AuthenticationState> = PublishSubject.create()
+    val authState: Observable<AuthenticationState> get() = _state
+
+    companion object {
+        val shared = FingerprintStore()
+        private const val ANDROID_KEY_STORE = "AndroidKeyStore"
+        private const val DEFAULT_KEY = "DEFAULT_KEY"
+    }
+
+    sealed class AuthenticationState {
+        object Succeeded : AuthenticationState()
+        object Failed : AuthenticationState()
+        data class Error(val error: String?) : AuthenticationState()
+    }
+
+    init {
+        dispatcher.register.filterByType(FingerprintSensorAction::class.java)
+            .doOnDispose { stopListening() }
```

Comment:
Nice.

Reply:
I wander if we can do away with the `Start` and `Stop` actions if we can `doOnDispose()` and `doOnSubscribe()`.

Please file a `tech-debt`/`chore` issue.

Reply:
Filed as #166.

No further action needed here.